static/js/people.js

```diff
@@ -287,7 +287,10 @@ export function get_user_time(user_id) {
     const user_pref = get_user_time_preferences(user_id);
     if (user_pref) {
         const current_date = utcToZonedTime(new Date(), user_pref.timezone);
-        return format(current_date, user_pref.format, {timeZone: user_pref.timezone});
+        if (current_date instanceof Date && !Number.isNaN(current_date)) {
```

Comment:
Why do we need both conditionals here? Also, surely we can look at `utcToZonedTime` and understand what types of other values it might return? (I think the most likely story here is the `timezone` was invalid).

Here is that function, from `zulip/node_modules/date-fns-tz/utcToZonedTime/index.js`:

```
function utcToZonedTime(dirtyDate, timeZone, options) {                                                  
  var date = (0, _index2.default)(dirtyDate, options);                                                   
  var offsetMilliseconds = (0, _index.default)(timeZone, date, true);                                    
  var d = new Date(date.getTime() - offsetMilliseconds);                                                 
  var resultDate = new Date(0);                                                                          
  resultDate.setFullYear(d.getUTCFullYear(), d.getUTCMonth(), d.getUTCDate());                           
  resultDate.setHours(d.getUTCHours(), d.getUTCMinutes(), d.getUTCSeconds(), d.getUTCMilliseconds());    
  return resultDate;                                                                                     
}                                                                                                        
```

If we can do `console.log` debugging with the user or something, it'd be very helpful to figure out what is failing here.

Reply:
Ahh, I see you actually did investigate this; we should add a comment for this workaround linking to https://github.com/marnusw/date-fns-tz/issues/191, since that's the upstream bug involved.

(In general, it's super helpful to include that sort of information in the comments, since this is a check that on the face of it shouldn't be required, and that issue is the answer to the natural question of "why is this required?")

Reply:
Still, why we do we need to check both the `instanceof` check and the `Number.isNaN` check? Does `NaN` pass the first check?

Reply:
We need to check `instanceof Date` because we know `utcToZonedTime` should return `Date` object, but is returning `undefined` here due to some bug. Then, to verify that date is real and not `Date("foo")` we need to check `!isNaN`.

Reply:
I think https://github.com/marnusw/date-fns-tz/issues/191 is more related to the part of why `utcToZonedTime` is returning undefined on our desktop app. I am not sure if it is the cause.

Reply:
> If we can do console.log debugging with the user or something, it'd be very helpful to figure out what is failing here.

User doesn't have a dev setup for us to do that.

Reply:
I tested this, and I think this `NaN` check is always false:
```
Number.isNaN(new Date("goo"))
false
```

Reply:
<img width="126" alt="image" src="https://user-images.githubusercontent.com/25124304/203613447-e58746d3-3549-4c76-871c-b181905668aa.png">
haha!

Reply:
You are looking for `Number.isNaN(x.getTime())`.

- #23659